### PR TITLE
Command deduplication - test case that uses CommandService and CommandSubmissionService [KVL-1106]

### PR DESCRIPTION
**Pull request by nicu-da**
- randomly use either the CommandService or CommandSubmissionService to test deduplication

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala [hunk 1]
```diff
@@ -192,6 +193,56 @@ private[testtool] abstract class CommandDeduplicationBase(
...
+          if (submitAndWait)
+            submitAndWaitRequestAndAssertDeduplication(ledger)(submitAndWaitRequest)
+          else submitRequestAndAssertDeduplication(ledger)(submitRequest, party)
+        }
+
+        deduplicationWait =>
+          for {
+            // Submit command (first deduplication window)
+            _ <- submitAndAssertAccepted(firstSubmitAndWait)
+            _ <- submitAndAssertDeduplicated(firstSubmitAndWait)
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala:228]: This doesn't seem to test the cases where we use one endpoint, then the other. For example, submit with the CommandService, and then check that the CommandSubmissionService deduplicates, and vice versa.

I suggest we just generate 4 random booleans, and run the test 10 times.

Comment (SamirTalwar-DA) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala:228]: (Or systematically test all 16 possibilities.)

Comment (nicu-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala:228]: Ok, I will just test all the possibilities in parallel

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala [hunk 2]
```diff
@@ -192,6 +192,69 @@ private[testtool] abstract class CommandDeduplicationBase(
...
 
+  // Without the submission id we cannot assert the received completions for parallel submissions
+  if (deduplicationFeatures.appendOnlySchema)
+    testGivenAllParticipants(
+      s"${testNamingPrefix}SimpleDeduplicationMixedClients",
+      "Deduplicate commands within the deduplication time window using the command client and the command submission client",
+      allocate(SingleParty),
+    )(implicit ec =>
+      configuredParticipants => { case Participants(Participant(ledger, party)) =>
+        def combinations(elements: List[List[Boolean]]): List[List[Boolean]] = elements match {
```
Comment (hubert-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala:203]: 🚲 IIRC, this is a variation with repetitions, so to be more precise I would call it `variations`.

**Review by hubert-da — APPROVED** [review]
LGTM

**Pull request merged**